softkinetic_camera/src/softkinetic_start.cpp

```diff
@@ -1016,6 +1020,32 @@ int main(int argc, char* argv[])
     {
       device_index = atoi(argv[1]);
     }
+    else
+    {
+      nh.param<bool>("use_serial", use_serial, false);
+      if (use_serial)
+      {
+        nh.getParam("serial", serial);
```

Comment:
if `use_serial` is `true` but `serial` not specified, or in the wrong namespace, we only get the Error "Serial number device not found".

can you please check the `serial` string and add a proper error output?
Maybe it would make sense to add the `serial` to the Error outputs below as well.